open-api/rest-catalog-open-api.yaml

```diff
@@ -2838,6 +2988,69 @@ components:
           additionalProperties:
             type: string
 
+    PreplanTableRequest:
+      type: object
+      required:
+        - select
+        - filter
+      properties:
+        select:
+          description: A list of the selected column names. If referencing a column with a nested field, ensure full name is joined by `.` such as `a.b`
+          type: array
+          items:
+            type: string
+        filter:
+          $ref: '#/components/schemas/Expression'
+        case-sensitive:
+          description: Indicates whether column selection and filtering should be case sensitive
+          type: boolean
+          default: true
+        snapshot-id:
+          description: Defaults to the table's current snapshot; This option is not allowed when `snapshot-range` is present in the request.
+          type: integer
+          format: int64
+        snapshot-range:
+          description: A JSON list containing exactly 2 snapshot IDs representing the start (exclusive) and end (inclusive) snapshots. This option is not allowed when `snapshot-id` is present in the request.
+          type: array
+          items:
+            type: integer
+            format: int64
+
+    PlanTableRequest:
+      type: object
+      required:
+        - select
+      properties:
+        select:
+          description: A list of the selected column names
+          type: array
+          items:
+            type: string
+        filter:
+          $ref: '#/components/schemas/Expression'
+        case-sensitive:
+          description: Indicates whether column selection and filtering should be case sensitive
+          type: boolean
+          default: true
+        stats-fields:
+          description: A list of field names for which stats should be included
+          type: array
+          items:
+            type: string
+        plan-task:
+          $ref: '#/components/schemas/PlanTask'
+        snapshot-id:
+          description: defaults to the table's current snapshot. This option is not allowed when `snapshot-range` or 'plan-task` is present in the request.
+          type: integer
+          format: int64
+
+        snapshot-range:
+          description: A JSON list containing exactly 2 snapshot IDs representing the start (exclusive) and end (inclusive) snapshots. This option is not allowed when `snapshot-id` or `plan-task` is present in the request.
+          type: array
```

Comment:
It seems a little awkward that we're using an array for this.  Why not a component with just a `fromExclusive` and `toInclusive` snapshot ids.

Reply:
This is something we could do. I suggested the range array but I called out that we may just want to have start and end: https://github.com/apache/iceberg/pull/9695#discussion_r1494914340

If you prefer the other option then let's go with that.